File: SimulationJava/src/Jagody.java
import javax.swing.ImageIcon;
import javax.swing.JLabel;

public class Jagody extends Roslina {

    Jagody(int xx,int yy, Swiat s){
        x=xx;
        y=yy;
        inicjatywa=0;
        sila=99;
        this.s=s;
        if(s instanceof Hexagony) obrazek = new JLabel(new ImageIcon("jagodyhex.png"));
        else obrazek = new JLabel(new ImageIcon("wilczejagody.png"));
    }

    @Override
    protected void kolizja(int xx,int yy){
        zycie = 0;
        makeLog(s.getPole(xx, yy).wypiszNazwe(), wypiszNazwe(), 3, x, y);
        s.getPole(xx, yy).setZycie(0);
        s.setTab(x, y, null);
        s.setTab(xx, yy, null);
    }

    @Override
    public Organizm dziecko() {
        return new Jagody(x,y,s);
    }

    @Override
    public String wypiszNazwe() {
        return "wilcze jagody";
    }

    
    @Override
    public JLabel wypisz() {
        return obrazek;
    }
    
}

File: SimulationCpp/projektPO/jagody.cpp
#include "jagody.h"
#include <iostream>
using namespace std;

Jagody::Jagody(int x, int y, Swiat* s) {
	sila = 99;
	this->x = x;
	this->y = y;
	zycie = ZYWY;
	this->s = s;
}

bool Jagody::probaRozsiania() {
	int i = rand() % SZANSANAROZSIANIEJAGOD;
	if (i == UDANA)return true;
	return false;
}

void Jagody::wypisz()const {
	SetConsoleTextAttribute(GetStdHandle(STD_OUTPUT_HANDLE), KOLORJAGOD);
	char a = SYMBOLJAGOD;
	cout << a;
}

void Jagody::kolizja(int nX, int nY) {
	zycie = MARTWY;
	s->tab[nX][nY]->zycie = MARTWY;
	makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(x), to_string(y), KOMr4);
	s->tab[nX][nY] = nullptr;
	s->tab[x][y] = nullptr;
}

string Jagody::wypiszNazwe() const {
	return "wilcze jagody";
}
Organizm* Jagody::dziecko() {
	return new Jagody(x, y, s);
}
